system-addon/test/unit/lib/init-store.test.js

```diff
@@ -61,6 +62,30 @@ describe("initStore", () => {
     store.dispatch({type: "FOO"});
     assert.notCalled(global.sendAsyncMessage);
   });
+  it("should be able to add custom properties within middleware", () => {
```

Comment:
Not really sure if this test is necessary. Would be a bit similar to having a test that javascript `+` can add two numbers -- i.e., it should just work. And if it doesn't work, other things will be broken, e.g., the other test you added for `assert.equal(store._earlyMessageQueue.length, 1);`

Reply:
You're right, perhaps adding some comments in init-store.jsm is better for now. Eventually, this could be fixed by #3725